### PR TITLE
Added Unit Tests For Configuration Module

### PR DESCRIPTION
**Pull request by zanpizmoht**
Signed-off-by: zanpizmoht <zan.pizmoht@comtrade.com>

In this PR unit tests for Configuration module are added.

**Related Issue**
None

**Description of the solution adopted**
I added unit tests for all the classes that are in commons/configuration module.

### CODE REVIEW

File: commons/src/test/java/org/eclipse/kapua/commons/configuration/KapuaConfigurableServiceSchemaUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    @Test
+    public void dropSchemaObjectsWithEmptyPathTest() {
+        KapuaConfigurableServiceSchemaUtils.dropSchemaObjects("");
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void dropSchemaObjectsWithNullPathTest() {
+        KapuaConfigurableServiceSchemaUtils.dropSchemaObjects(null);
+    }
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/configuration/KapuaConfigurableServiceSchemaUtilsTest.java:161]: Please add additional tests for DropSchemaObjects method (for now we have default, empty, null and wrong, but we also need some custom strings to be tested).

File: commons/src/test/java/org/eclipse/kapua/commons/configuration/KapuaConfigurableServiceSchemaUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    @Test
+    public void createSchemaObjectsWithEmptyPathTest() throws KapuaException {
+        KapuaConfigurableServiceSchemaUtils.createSchemaObjects("");
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void createSchemaObjectsWithNullPathTest() throws KapuaException {
+        KapuaConfigurableServiceSchemaUtils.createSchemaObjects(null);
+    }
+
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/configuration/KapuaConfigurableServiceSchemaUtilsTest.java:118]: Please add additional tests for CreateSchemaObjects method (for now we have default, empty, null and wrong, but we also need some custom strings to be tested)

File: commons/src/test/java/org/eclipse/kapua/commons/configuration/ServiceComponentConfigurationImplTest.java [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        assertNull(serviceComponentConfiguration.getId());
+    }
+
+    @Test
+    public void setIdToLargeValueTest() {
+        serviceComponentConfiguration.setId("12312312312312331232313123123123");
+        assertEquals(serviceComponentConfiguration.getId(), "12312312312312331232313123123123");
+    }
+
+    @Test
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/configuration/ServiceComponentConfigurationImplTest.java:64]: I think this test is obsolete, it is basically a copy of:

   @Test
    public void getIdIfIsSetTest() {
        serviceComponentConfiguration.setId("1");
        assertEquals(serviceComponentConfiguration.getId(), "1");
    }

File: commons/src/test/java/org/eclipse/kapua/commons/configuration/ServiceComponentConfigurationImplTest.java [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+        assertEquals(serviceComponentConfiguration.getName(), "name");
+    }
+
+    @Test
+    public void setNameToNullTest() {
+        serviceComponentConfiguration.setName(null);
+        assertNull(serviceComponentConfiguration.getName());
+    }
+
+    @Test
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/configuration/ServiceComponentConfigurationImplTest.java:93]: I think this test is obsolete, it is basically a copy of:
    @Test
    public void setNameIfNotNullTest() {
        serviceComponentConfiguration.setName("name");
        assertEquals(serviceComponentConfiguration.getName(), "name");
    }

File: commons/src/test/java/org/eclipse/kapua/commons/configuration/ServiceComponentConfigurationImplTest.java [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+            public Collection<Object> values() {
+                return null;
+            }
+
+            @Override
+            public Set<Entry<String, Object>> entrySet() {
+                return null;
+            }
+        };
+        properties.put("property1", 10);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/configuration/ServiceComponentConfigurationImplTest.java:127]: Cna we please try with different objects rather than just Integer?

**Review by LeoNerdoG — CHANGES_REQUESTED** [review]
Hey @zanpizmoht,
I have reviewed this branch and there are some minor things to be fixed. 
Please rebase this PR on develop, as it is not up-to-date.

**Comment by zanpizmoht** [comment]
Hey @LeoNerdoG, I have modified tests as you suggested and it should now be ready for merge :)

**Review by LeoNerdoG — APPROVED** [review]

**Review by Coduz — APPROVED** [review]
